src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,219 @@
+  const [isEditing, setIsEditing] = useState(false);
+  const [editedTodoTitle, setEditedTodoTitle] = useState(title);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (isEditing) {
+      inputRef.current?.focus();
+    }
+  }, [isEditing]);
+
+  useEffect(() => {
+    setIsCompleted(completed);
+    setEditedTodoTitle(title);
+  }, [completed, title]);
+
+  function handleDeleteTodo(todoId: number) {
+    setIsLoading(true);
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+      })
+      .catch(() => {
+        setErrorMessage(Error.unableToDelete);
+        setTimeout(() => {
+          setErrorMessage(Error.none);
+        }, 3000);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }
+
+  function handleToggleTodo(updatedTodo: Todo) {
+    setIsLoading(true);
+    updateTodo(updatedTodo)
+      .then(todo => {
+        setTodos(currentTodos => {
+          const newTodos = [...currentTodos];
+          const index = newTodos.findIndex(item => item.id === updatedTodo.id);
+
+          newTodos.splice(index, 1, todo);
+
+          return newTodos;
+        });
+        setIsCompleted(!isCompleted);
+      })
+      .catch(() => {
+        setErrorMessage(Error.unableToUpdate);
+        setTimeout(() => {
+          setErrorMessage(Error.none);
+        }, 3000);
+      })
+      .finally(() => setIsLoading(false));
+  }
+
+  function handleEditTodo(updatedTodo: Todo) {
+    setIsLoading(true);
+    if (editedTodoTitle.trim() === title) {
+      setIsEditing(false);
+      setIsLoading(false);
+
+      return;
+    }
+
+    if (editedTodoTitle.trim() === '') {
+      handleDeleteTodo(id);
+
+      return;
+    }
+
+    updateTodo(updatedTodo)
+      .then(todo => {
+        setTodos(currentTodos => {
+          const newTodos = [...currentTodos];
+          const index = newTodos.findIndex(item => item.id === updatedTodo.id);
+
+          newTodos.splice(index, 1, todo);
+
+          setIsEditing(false);
+
+          return newTodos;
+        });
+      })
+      .catch(() => {
+        inputRef.current?.focus();
+        setErrorMessage(Error.unableToUpdate);
+        setTimeout(() => {
+          setErrorMessage(Error.none);
+        }, 3000);
+      })
+      .finally(() => setIsLoading(false));
+  }
+
+  return (
+    <div
+      key={id}
+      data-cy="Todo"
+      className={classNames('todo', { completed: completed })}
+    >
+      <label className="todo__status-label">
+        {' '}
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={completed}
+          onChange={() =>
+            handleToggleTodo({
+              id: id,
+              title: title,
+              completed: !isCompleted,
+              userId: USER_ID,
+            })
```

Comment:
Move this logic to the handler function

Reply:
Not fixed 
<img width="784" alt="image" src="https://github.com/user-attachments/assets/d1fcd0d0-76cf-4b23-be19-4e57017a4416">